docs/src/pages/components/rating/demo.tsx

```diff
@@ -1,52 +1,42 @@
 import React from 'react';
-import { Rating } from '@aws-amplify/ui-react';
-import { RatingPropControls } from '@/components/RatingPropControls';
-import { Example } from '@/components/Example';
-
-export const RatingDemo = () => {
-  const [fillColor, setFillColor] = React.useState<string>();
-  const [emptyColor, setEmptyColor] = React.useState<string>();
-  const [size, setSize] = React.useState();
-  const [value, setValue] = React.useState<number>(0);
-  const [maxValue, setMaxValue] = React.useState<number>(5);
-
-  const ratingProps = {
-    size,
-    setSize,
-    value,
-    setValue,
-    maxValue,
-    setMaxValue,
-    fillColor,
-    setFillColor,
-    emptyColor,
-    setEmptyColor,
-  };
+import { Rating, RatingProps } from '@aws-amplify/ui-react';
+import { RatingPropControls } from './RatingPropControls';
+import { useRatingProps } from './useRatingProps';
+import { Demo } from '@/components/Demo';
 
+const propsToCode = (props) => {
   return (
-    <>
-      <RatingPropControls {...ratingProps} />
-      <Example>
-        <Rating
-          fillColor={fillColor}
-          emptyColor={emptyColor}
-          size={size}
-          value={value}
-          maxValue={maxValue}
-        />
-      </Example>
-    </>
+    `<Rating` +
+    (props.size ? `\n  size=${JSON.stringify(props.size)}` : '') +
+    `
+  value={${props.value}}
+  maxValue={${props.maxValue}}
+  fillColor="${props.fillColor}"
+  emptyColor="${props.emptyColor}"
+   />`
   );
 };
 
-export const MyCoolSvg = () => {
+export const RatingDemo = () => {
+  const ratingProps = useRatingProps({
+    value: 0,
```

Comment:
Let's set the value to something other than zero so that the demo displays on initial load. 

This also brings up a question about an edge case: If the value is zero, I would expect the Rating to display empty stars by default instead of no stars.